extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+				return $true
+			}
+			else {
+				Write-OutputAndLog "Failed to verify $packageName installation."
+				return $false
+			}
+		}
+		catch {
+			Write-ErrorAndLog -exception $_.Exception
+			# Only throw the error incase there is a valid verification command
+			if (-not [string]::IsNullOrEmpty($installationVerificationCallback.ToString())) {
+				Write-ErrorAndLog "Installation Error: Installation of package $packageName failed verification"
+			}
+			return $false
+		}
+	}
+	catch {
+		Write-ErrorAndLog -exception $_.Exception
+		return $false
+	}
+	return $true
+}
+
+#################################### Register Node to DMS ###################################
+Function Register-IntegrationRuntime {
+	<#
+    .DESCRIPTION
+    Configures the SHIR with the auth key. It registers the IR to the DMS using the authentication keys.
+
+    .OUTPUTS
+    None.
+    #>
+	param (
+		[Parameter()]
+		[string]$AuthKey1,
+
+		[Parameter()]
+		[string]$AuthKey2
+	)
+
+	if ([string]::IsNullOrWhitespace($AuthKey1) -and [string]::IsNullOrWhitespace($AuthKey2)) {
+		Write-ErrorAndLog "You need to provide atleast one authentication key to configure SHIR."
+		return
+	}
+
+	$azDataMigrationModule = Get-Module -ListAvailable -Name Az.DataMigration
+
+	if (-not $azDataMigrationModule) {
+		# Install the NugetProvider for Az.DataMigration
+		$nuget = Get-PackageProvider -Name NuGet -ErrorAction SilentlyContinue
+
+		if ($null -eq $nuget -or $nuget.Version -lt $Global:MinRequiredNuGetVersion) {
+			Install-PackageProvider -Name NuGet -MinimumVersion $Global:MinRequiredNuGetVersion -Force
+		}
+
+		# Install the Az.DataMigration module
+		Install-Module -Name Az.DataMigration -Force
+	}
+
+	# Try the first authentication key
+	try {
+		if (-not [string]::IsNullOrWhitespace($AuthKey1)) {
+			Write-OutputAndLog "Trying the first authentication key..."
+			Register-AzDataMigrationIntegrationRuntime -AuthKey $AuthKey1 *> $null
+			Write-OutputAndLog "Registration successful with the first authentication key."
+			return
+		}
+	}
+	catch {
+		Write-ErrorAndLog "Failed to register with the first authentication key. Error: $($_.Exception.Message)"
+	}
+
+	# Try the second authentication key
+	try {
+		if (-not [string]::IsNullOrWhitespace($AuthKey2)) {
+			Write-OutputAndLog "Trying the second authentication key..."
+			Register-AzDataMigrationIntegrationRuntime -AuthKey $AuthKey2 *> $null
+			Write-OutputAndLog "Registration successful with the second authentication key."
+			return
+		}
+	}
+	catch {
+		Write-ErrorAndLog "Failed to register with the second authentication key. Error: $($_.Exception.Message)"
+	}
+}
+
+###################################### Main Execution #####################################
+if (-not (Test-AdminAccess)) {
+	if ($AdminPriv) {
+		Write-ErrorAndLog -exception "Failed to gain admin privileges."
+	}
+	# Gain admin access if script is executed in a without admin privileges by prompting the user
```

Comment:
Please update the comment as few words are missing in it

Reply:
Updated